src/main/java/org/rumbledb/expressions/flowr/ForClauseVar.java

```diff
@@ -35,13 +35,13 @@
 public class ForClauseVar extends FlworVarDecl {
 
     private final boolean allowEmpty;
-    private final VariableReference positionalVariableReference;
+    private final VariableReferenceExpression positionalVariableReference;
 
     public ForClauseVar(
-            VariableReference varRef,
+            VariableReferenceExpression varRef,
             FlworVarSequenceType seq,
             boolean emptyFlag,
-            VariableReference atVarRef,
+            VariableReferenceExpression atVarRef,
```

Comment:
full names.

Shall we rename atVarRef here to positionalVariableReference for clarity?